### PR TITLE
EG: Notebook For Prediction Of Molecular Pathways & Key Mutations

### PR DESCRIPTION
**Pull request by simongraham**
Added example notebook for mutation prediction. Include the 2 stage IDaRS pipeline to predict key mutations and molecular pathways. First, tumour detection is performed and then the WSI-level prediction task is done. WSI level prediction tasks considered:

- MSI 
- TP53
- BRAF
- CIN
- CIMP
- HM

As part of this PR, we retrained IDaRS models without stain normalisation to speed up inference.

### CODE REVIEW

**Review by DavidBAEpstein — CHANGES_REQUESTED** [review]
See ReviewNB. Please fix code so that it can be run in Colab. (Admittedly, I haven't tried to run it, but it seems to me that it wouldn't run.)
There needs to be some advice on to how to arrange for GPU use on Colab, possibly also on other platforms. I will carry on reviewing once the notebook runs on Colab (without GPU), and you can respond to comments about text cells at your leisure, for example waiting until I have gone through the whole notebook once.

**Review by DavidBAEpstein — CHANGES_REQUESTED** [review]
Although 8 files have been changed, I looked only at the 07 jupyter notebook.
I found two independent fatal errors. I have made comments in ReviewNB. The first error was obvious, so I just commented it out and tried running the notebook under Colab. The error message from the second fatal error is given in full in ReviewNB.

File: tiatoolbox/models/engine/patch_predictor.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # along with this program; if not, write to the Free Software Foundation,
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 #
-# The Original Code is Copyright (C) 2021, TIA Centre, University of Warwick
+# The Original Code is Copyright (C) 2021, TIALab, University of Warwick
```
Comment (shaneahmed) [inline tiatoolbox/models/engine/patch_predictor.py:17]: It should be `TIA Centre`

**Review by DavidBAEpstein — COMMENTED** [review]
The grammar is fine. In the last cell "We welcome you to ..."  is unusual, but OK. More usual would be "You are welcome to ....".
The notebook seems to be in better shape than when I last looked at it. But it still depends on Dang's personal file system, so it can't be merged yet. The notebook is not yet in a form where it can be run with GPU in Colab. I hope that it is the intention to fix that.

**Review by DavidBAEpstein — COMMENTED** [review]
I see nothing wrong with the grammar of the 08 notebook. The objections I had previously to merging remain, namely that 1) the merged version must avoid using Dang's private files, and 2) the notebook should be set up to run a GPU under Colab. The final text cell could be phrased in a more usual form as "You are welcome to ..." rather than "We welcome you to ..." but it's acceptable in the current form.

**Review by DavidBAEpstein — CHANGES_REQUESTED** [review]
I saved your notebook in HTML format (using jupyter notebook > File > Download As) and then used Track format (TRK) in MS Word to record suggested changes---there were not many of these. Please use TRK to find my suggested changes. The conversion process introduced a few errors. I should try using nbconvert.
[08_example_mutationprediction.docx](https://github.com/TissueImageAnalytics/tiatoolbox/files/7613188/08_example_mutationprediction.docx)
s.

File: setup.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
     "zarr",
     "requests",
     "defusedxml",
+    "pygeos",
```
Comment (shaneahmed) [inline setup.py:34]: this is being removed. Please make sure you update this after #202 is merged.

**Review by DavidBAEpstein — CHANGES_REQUESTED** [review]
[DE_TRK_01_idars.docx](https://github.com/TissueImageAnalytics/tiatoolbox/files/7692908/DE_TRK_01_idars.docx)
please use "next" in Word > Review > Track changes
to find all the suggestions (of which there are very few).
Feel free to ignore suggestions that you don't agree with.

**Comment by simongraham** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/TissueImageAnalytics/tiatoolbox/pull/189/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by simongraham** [comment]
<p>Sorted</p>

---

 <a href='https://app.reviewnb.com/TissueImageAnalytics/tiatoolbox/pull/189/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by simongraham** [comment]
<p>Done</p>

---

 <a href='https://app.reviewnb.com/TissueImageAnalytics/tiatoolbox/pull/189/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Pull request merged**